packages/peregrine/lib/talons/WishlistPage/useCreateWishlist.js

```diff
@@ -1,12 +1,45 @@
-import { useState, useCallback } from 'react';
+import { useState, useCallback, useMemo } from 'react';
+import { useMutation, useApolloClient } from '@apollo/client';
+import mergeOperations from '../../util/shallowMerge';
+
+import DEFAULT_OPERATIONS from './createWishlist.gql';
 
 /**
  * @function
  *
  * @returns {CreateWishListProps}
  */
-export const useCreateWishlist = () => {
+export const useCreateWishlist = (props = {}) => {
+    const operations = mergeOperations(DEFAULT_OPERATIONS, props.operations);
+    const { createWishlistMutation, getCustomerWishlistsQuery } = operations;
+    const apolloClient = useApolloClient();
     const [isModalOpen, setIsModalOpen] = useState(false);
+    const [createWishlist, { error: createWishlistError }] = useMutation(
+        createWishlistMutation,
+        {
+            update(
+                cache,
+                {
+                    data: { createWishlist }
+                }
+            ) {
+                const { customer } = apolloClient.readQuery({
+                    query: getCustomerWishlistsQuery
+                });
+                apolloClient.writeQuery({
```

Comment:
This manual cache write ends up just triggering the wishlistPage query to refetch, which is a lot more straight forward code wise, and you have the ability to stall completion of the mutation until the refetch comes back. Consider it optional since this works and that refetch doesn't appear to cause any re-render, but you can find an example of that [in the addressBook component](https://github.com/magento/pwa-studio/blob/870cb896919f67effe21ef37f2f498867b2193cf/packages/peregrine/lib/talons/AddressBookPage/useAddressBookPage.js#L121).

Reply:
Thank you for advice! Changed